be/src/exec/vectorized/join_hash_map.tpp

```diff
@@ -45,6 +45,49 @@ Status JoinBuildFunc<PT>::construct_hash_table(RuntimeState* state, JoinHashTabl
     return Status::OK();
 }
 
+template <PrimitiveType PT>
+void DirectMappingJoinBuildFunc<PT>::prepare(RuntimeState* runtime, JoinHashTableItems* table_items) {
+    table_items->bucket_size =
+            (int64_t)(std::numeric_limits<CppType>::max()) - (int64_t)(std::numeric_limits<CppType>::min() + 1l);
+    table_items->first.resize(table_items->bucket_size, 0);
+    table_items->next.resize(table_items->row_count + 1, 0);
+}
+
+template <PrimitiveType PT>
+const Buffer<typename DirectMappingJoinBuildFunc<PT>::CppType>& DirectMappingJoinBuildFunc<PT>::get_key_data(
+        const JoinHashTableItems& table_items) {
+    if (table_items.key_columns[0]->is_nullable()) {
+        auto* null_column = ColumnHelper::as_raw_column<NullableColumn>(table_items.key_columns[0]);
+        return ColumnHelper::as_raw_column<ColumnType>(null_column->data_column())->get_data();
+    }
+
+    return ColumnHelper::as_raw_column<ColumnType>(table_items.key_columns[0])->get_data();
+}
+
+template <PrimitiveType PT>
+Status DirectMappingJoinBuildFunc<PT>::construct_hash_table(RuntimeState* state, JoinHashTableItems* table_items,
+                                                            HashTableProbeState* probe_state) {
+    auto& data = get_key_data(*table_items);
+    if (table_items->key_columns[0]->is_nullable()) {
+        auto* nullable_column = ColumnHelper::as_raw_column<NullableColumn>(table_items->key_columns[0]);
+        auto& null_array = nullable_column->null_column()->get_data();
+        for (size_t i = 1; i < table_items->row_count + 1; i++) {
+            if (null_array[i] == 0) {
+                auto buckets = data[i] - std::numeric_limits<CppType>::min();
```

Comment:
define a constexpr instead of std::numeric_limits<CppType>::min()
use size_t instead of auto

Reply:
Fixed